streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -188,6 +193,20 @@ void flushState() {
     }
 
     void initializeStateStores() {
+        if (topology.stateStores().isEmpty()) {
```

Comment:
Actually, that has a pretty big knock on effect as we have to check that we have a state manager everywhere. We also pass it into `ProcessorContextImpl` so we'd have to check in places it is used there, too. Seems very smelly